graphql2/graphqlapp/alertlog.go

```diff
@@ -0,0 +1,129 @@
+package graphqlapp
+
+import (
+	"context"
+	"github.com/pkg/errors"
+	"github.com/target/goalert/alert"
+	"github.com/target/goalert/alert/log"
+	"github.com/target/goalert/graphql2"
+	"github.com/target/goalert/search"
+	"github.com/target/goalert/validation/validate"
+)
+
+type AlertLog App
+
+func (a *App) AlertLog() graphql2.AlertLogResolver { return (*AlertLog)(a) }
+
+func (l *AlertLog) Event(ctx context.Context, log *alert.Log) (graphql2.Event, error) {
+	return graphql2.Event(log.Event), nil
+}
+
+func (l *AlertLog) AlertID(ctx context.Context, log *alert.Log) (int, error) {
+	var s alertlog.SearchOptions
+	s.Start = log.Timestamp
+	s.Event = alertlog.Type(log.Event)
+	s.Limit = 1
+
+	entries, _, err := l.LogStore.Search(ctx, &s)
+	if err != nil {
+		return -1, err
+	}
+	return entries[0].AlertID(), nil
+}
+
+func (q *Query) Alertlogs(ctx context.Context, opts *graphql2.AlertLogSearchOptions) (conn *graphql2.AlertLogConnection, err error) {
+	if opts == nil {
+		opts = &graphql2.AlertLogSearchOptions{}
+	}
+
+	var s alertlog.SearchOptions
+
+	if opts.After != nil && *opts.After != "" {
+		err = search.ParseCursor(*opts.After, &s)
+		if err != nil {
+			return nil, errors.Wrap(err, "parse cursor")
+		}
+	}
+
+	if opts.FilterByAlertID != nil {
+		s.AlertID = *opts.FilterByAlertID
+	} else {
+		s.AlertID = 0
+	}
+
+	if opts.FilterByServiceID != nil {
+		s.ServiceID = *opts.FilterByServiceID
+	}
+
+	s.SortDesc = *opts.SortDesc
+
+	if opts.Event != nil {
+		switch *opts.Event {
+		case graphql2.EventCreated:
+			s.Event = "created"
+		case graphql2.EventClosed:
+			s.Event = "closed"
+		case graphql2.EventNotificationSent:
```

Comment:
Can we map `alertlog.Type` into graphql directly (inside `gqlgen.yml`) instead of using strings here?